ecr-login/api/factory.go

```diff
@@ -22,6 +22,7 @@ import (
 
 type ClientFactory interface {
 	NewClient(awsSession *session.Session, awsConfig *aws.Config) Client
+	NewClientWithCacheDir(awsSession *session.Session, awsConfig *aws.Config, cacheDir string) Client
```

Comment:
I wonder if it's worth making a `Options` struct to capture these parameters and then have `NewClientWithOptions`?

```go
type Options struct {
	Session  *session.Session
	Config   *aws.Config
	CacheDir string
}
```